src/Aspire.Hosting.DevTunnels/DevTunnelResourceBuilderExtensions.cs

```diff
@@ -440,7 +442,8 @@ public static IResourceBuilder<TResource> WithReference<TResource>(this IResourc
 
                     if (flags.HasFlag(ReferenceEnvironmentInjectionFlags.Endpoints))
                     {
-                        context.EnvironmentVariables[$"{serviceName.ToUpperInvariant()}_{endpointName.ToUpperInvariant()}"] = port.TunnelEndpoint;
+                        var endpointKey = $"{encodedServiceName.ToUpperInvariant()}_{encodedEndpointName.ToUpperInvariant()}";
+                        context.EnvironmentVariables[endpointKey] = port.TunnelEndpoint;
```

Comment:
Is it ever valid to have a non alphanumeric/underscore in an env var name? The problem with requiring the user to use EnvironmentVariableNameEncoder is they'll inevitably not know encoding is required or forget.

I wonder if the encoding should happen at a lower level. Why the current approach?

Reply:
I just spoke to Seb about that.